### PR TITLE
update CHANGELOG for 5.0.5

### PR DESCRIPTION
**Pull request by keegancsmith**
I went through the list of cherry-picks and moved items from unreleased into 5.0.5. I believe the fix code-intel popovers was incorrectly put under the 5.0.4 changelog.

Test Plan: n/a

Part of https://github.com/sourcegraph/sourcegraph/issues/52745

### CODE REVIEW

**Comment by keegancsmith** [comment]
Ping @taras-yemets to confirm I correctly moved his changelog entry.

**Comment by keegancsmith** [comment]
@olafurpg @eseliger @erzhtor I believe there are entries missing for these cherry-picks. Please push to this branch or main or comment why it isn't needed.

```
- olaf :: a1f6e1f88d [Backport 5.0] Bump syntax-highlighter docker container for single-docker deployment (#52712)
- erik :: a74f104a1a [Backport 5.0] Make endpoint configurable for OpenAI completions client (#52601)
- erik :: d7bb5a1aea Backport Implement features in LLM Proxy layer to 5.0 (#52531)
- erzhan :: d2eac1e79d [Backport 5.0] (feat) add ability to allow anonymous usage mode based on "auth.publi… (#52506)
```

**Review by kopancek — APPROVED** [review]

**Review by taras-yemets — APPROVED** [review]

**Review by erzhtor — APPROVED** [review]
Thanks! Added entry for cherry-pick: `2eac1e79d [Backport 5.0] (feat) add ability to allow...`

**Review by olafurpg — APPROVED** [review]
Done ✅

File: CHANGELOG.md [hunk 1]
```diff
@@ -73,6 +70,15 @@ All notable changes to Sourcegraph are documented in this file.
 ### Added
 
 - Organization members can now administer batch changes created by other members in their organization's namespace if the setting `orgs.allMembersBatchChangesAdmin` is enabled for that organization. [#50724](https://github.com/sourcegraph/sourcegraph/pull/50724)
+- Allow instance public access mode based on `auth.public` site config and `allow-anonymous-usage` license tag [#52440](https://github.com/sourcegraph/sourcegraph/pull/52440)
```
Comment (eseliger) [inline CHANGELOG.md:73]:
```suggestion
- Allow instance public access mode based on `auth.public` site config and `allow-anonymous-usage` license tag [#52440](https://github.com/sourcegraph/sourcegraph/pull/52440)
- The endpoint configuration field for completions is now supported by the OpenAI provider [#52530](https://github.com/sourcegraph/sourcegraph/pull/52530)
```

**Pull request merged**